Fce 1

florentin-campagne opened the pull request:
Amélioration du modèle

- Ajout d'id technique sur la table 'utilisateur'
- Modification du type 'REFERENCES livre' de la table 'tag' (SERIAL)
- Ajout des conditions de non nullité des colonnes (NOT NULL)

- Ajout de la précision de l'ordre des colonnes des tables pour les
INSERT INTO et ALTER TABLE dans la DAO

- Création de la classe Entity + modification des classes héritées
- Ajout des vérifications de nullité dans les setters (renvoie d'un
BeanException ??)
- Nettoyage des constructeurs
- Modification du nom du package 'entites' -> 'beans' (anglais)

- Déplacement du fichier init_base.sql dans le repertoire config/

Commit:
test

Commit:
Amélioration du modèle

- Ajout d'id technique sur la table 'utilisateur'
- Modification du type 'REFERENCES livre' de la table 'tag' (SERIAL)
- Ajout des conditions de non nullité des colonnes (NOT NULL)

- Ajout de la précision de l'ordre des colonnes des tables pour les
INSERT INTO et ALTER TABLE dans la DAO

- Création de la classe Entity + modification des classes héritées
- Ajout des vérifications de nullité dans les setters (renvoie d'un
BeanException ??)
- Nettoyage des constructeurs
- Modification du nom du package 'entites' -> 'beans' (anglais)

- Déplacement du fichier init_base.sql dans le repertoire config/

Commit:
Amélioration de la gestion des exceptions des DAO

- Création d'un DAOException
- catch des SQLException et NamingException // throw DAOException

Commit:
suppression des catch 'Exception'

fsabbe-adeo commented on config/init_base.sql:
```diff
@@ -0,0 +1,32 @@
+		prenom VARCHAR(50) NOT NULL);
+
+create table tag (
+		id_livre SERIAL REFERENCES livre NOT NULL, 
```
type INTEGER pour la FK

fsabbe-adeo commented on src/main/java/beans/Entity.java:
```diff
@@ -0,0 +1,15 @@
+		return id;
+	}
+
+	protected void setId(Integer id) {
```
public : Une entité peut exister sans être persistée (avoir un id issu de la BDD)

fsabbe-adeo commented on src/main/java/beans/Auteur.java:
```diff
@@ -0,0 +1,29 @@
+	public String getNom() {
+		return nom;
+	}
+	public void setNom(String nom) {
```
https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalArgumentException.html
idem pour tous les setters qui font ce controle
Les chaines vides ou remplies de caractères d'espacement sont valides ?

fsabbe-adeo commented on src/main/java/beans/Livre.java:
```diff
@@ -1,56 +1,54 @@
 	public ArrayList<String> getTags() {
 		return tags;
 	}
+	public void setTags(ArrayList<String> tags) {
```
pas d'accès public à cette propritété

fsabbe-adeo commented on src/main/java/dao/DAOException.java:
```diff
@@ -0,0 +1,12 @@
+	 */
+	private static final long serialVersionUID = 1L;
+
+	public DAOException(String message) {
```
Il faut surcharger tous les constructeurs de la classe Exception

Commit:
Amélioration du modèle

- remplacement 'SERIAL' -> 'INT' pour le REFERENCES de la table 'tag'
- public setId (Entity) (pour les entités non persistées)
- throw IllegalArgumentException dans les entités + ajout de contraintes
sur les chaines vides ou remplies d'espaces + début de javadoc
- private setTags (Livre)
- surcharge des constructeurs de la classe Exception dans DAOException
- ajout de la condition UNIQUE à la colonne 'login' de la table
'utilisateur'
- Modification de la classe 'Entity' en classe générique
'Entity<Object>'

+ début de correction des DAO (modification des fonctions afin de
prendre en compte des Entites plutôt que des champs)
+ traduction des noms des méthodes en anglais

fsabbe-adeo commented on src/main/java/beans/Auteur.java:
```diff
@@ -0,0 +1,55 @@
+	 * 		Si le parametre est null, constitue une chaine vide ou est remplie de caracteres d'espacements
+	 */
+	public void setNom(String nom) {
+		if (nom != null && !nom.equals("") && !nom.replaceAll("\\s+","").equals("")) {
```
Pas de if ... else pour les exceptions ici :
if(invalid){
  throw exception;
}
this.property = arg;

https://www.baeldung.com/java-blank-empty-strings
Utilise Apache Commons, ca te fera un exercice pour ajouter une dépendance avec maven comme vu ce matin

fsabbe-adeo commented on src/main/java/beans/Livre.java:
```diff
@@ -0,0 +1,83 @@
+	public ArrayList<String> getTags() {
+		return tags;
+	}
+	private void setTags(ArrayList<String> tags) {
```
protected, je t'expliquerai pourquoi quand on abordera les ORM

fsabbe-adeo commented on src/main/java/beans/Entity.java:
```diff
@@ -0,0 +1,17 @@
+	}
+
+	public void setId(Type id) {
+		if (id != null) {
```
Un id peut-être null => comme vu ce matin, cela indique que l'instance n'est pas persistée en BDD

fsabbe-adeo commented on src/main/java/beans/Auteur.java:
```diff
@@ -0,0 +1,55 @@
+		if (nom != null && !nom.equals("") && !nom.replaceAll("\\s+","").equals("")) {
+			this.nom = nom;
+		} else {
+			throw new IllegalArgumentException();
```
Toujours mettre un message pour préciser quelle est l'erreur

Commit:
Amélioration du modèle

- ajout de la dépendance Apache Commons à Maven
- Modification de la structure des exceptions dans les entites
- utilisation de isEmpty et isBlank (StringUtils)
- protected setTags (Livre)
- suppression de la condition dans setId (Entity)

Commit:
correction du pom

Commit:
supression des conditions inutiles des Entites
